stdlib/socket.pyi

```diff
@@ -549,26 +559,56 @@ class socket(_socket.socket):
     # Note that the makefile's documented windows-specific behavior is not represented
     # mode strings with duplicates are intentionally excluded
     @overload
+    def makefile(  # type: ignore[misc]
+        self,
+        mode: Literal["b", "rb", "br", "wb", "bw", "rwb", "rbw", "wrb", "wbr", "brw", "bwr"],
+        buffering: Literal[0],
+        *,
+        encoding: str | None = ...,
+        errors: str | None = ...,
+        newline: str | None = ...,
+    ) -> SocketIO: ...
+    @overload
     def makefile(
         self,
-        mode: Literal["r", "w", "rw", "wr", ""] = ...,
+        mode: Literal["rwb", "rbw", "wrb", "wbr", "brw", "bwr"],
         buffering: int | None = ...,
```

Comment:
This will pick the wrong overload when the user passes something that looks like an `int` to the typechecker but is `0` at runtime. For `builtins.open`, we handle that with a fallback overload that returns `BinaryIO`; we should do that here too.

Reply:
I'll use `IOBase` for the fallback for now as I'm interested what stubtest says. Another case for my pet peeve python/typing#566. :)

Reply:
@JelleZijlstra primer passes. I'd prefer to use the concrete `IOBase` here, instead of the `BinaryIO` pseudo-protocol. We can always change it later if someone complains. Maybe even returning a union here is alternative. Type checkers should treat it the same as `IOBase`, but it's closer to the implementation.

Reply:
Sure. We could try these approaches also with `open()`.